File: python/group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict = {}

        alpha_mapper = {
            "a": 0,
            "b": 1,
            "c": 2,
            "d": 3,
            "e": 4,
            "f": 5,
            "g": 6,
            "h": 7,
            "i": 8,
            "j": 9,
            "k": 10,
            "l": 11,
            "m": 12,
            "n": 13,
            "o": 14,
            "p": 15,
            "q": 16,
            "r": 17,
            "s": 18,
            "t": 19,
            "u": 20,
            "v": 21,
            "w": 22,
            "x": 23,
            "y": 24,
            "z": 25,
        }

        for word in strs:
            arr = [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
            ]
            for c in word:
                ind: int = alpha_mapper.get(c, 0)
                arr[ind] += 1

            arr_str = " ".join(map(str, arr))
            print(arr_str)
            val = dict.get(arr_str, [])
            val.append(word)
            dict[arr_str] = val
            print(dict.get(arr_str))

        return dict.values()

File: cpp/groupAnagrams.cpp
#include <string>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
  vector<vector<string>> groupAnagrams(vector<string> &strs) {
    unordered_map<string, vector<string>> m;

    for (const auto &str : strs) {
      vector<int> v(26, 0);

      for (char c : str) {
        v[c - 'a']++;
      }
      string key = to_string(v[0]);
      for (int i = 1; i < 26; i++) {
        key += ',' + to_string(v[i]);
      }
      m[key].push_back(str);
    }
    vector<vector<string>> res;
    for (const auto &entry : m) {
      res.push_back(entry.second);
    }
    return res;
  }
};
